stats/compute.go

```diff
@@ -0,0 +1,68 @@
+// Package stats provides a wrapper around github.com/montanaflynn/stats
+// to compute common statistics for a dataset.
+//
+// The package requires the dataset to be a slice of float64. The caller has
+// the responsibility of transforming the data into a usable dataset for
+// stats.Compute.
+//
+package stats
+
+import (
+	"fmt"
+
+	"github.com/montanaflynn/stats"
+)
+
+// Stats represents the statistics computed from a given dataset.
+type Stats struct {
+	Min    float64
+	Max    float64
+	Mean   float64
+	Median float64
+	StdDev float64
+
+	// Deciles is an array composed of the nine deciles.
+	// Deciles[0] is the first decile (10%), Deciles[8]
+	// is the 9th decile (90%).
+	Deciles [9]float64
+}
+
+// Compute computes the common statistics for a dataset. Returns ErrCompute
+// if any error occurs during the computation. When returning an error, the
+// value of Stats struct may be partially written.
+func Compute(data []float64) (Stats, error) {
+	if len(data) == 0 {
+		return Stats{}, ErrEmptySlice
+	}
+
+	out := Stats{}
+	errs := []string{}
+
+	out.Max, errs = pipe("max", errs)(stats.Max(data))
```

Comment:
So pretentious. I like it.

Reply:
👑 
Well its _pipes_ the output of the subsequent function call into its destination and any errs into the bucket of errors ahah. That or `handle`.

Reply:
Well now you suggest it, it's true that `pipe` has a specific meaning (especially in that functional style context) that does not correspond to what we do here... But that's not really a problem to me 🤷‍♂️

Reply:
I'm not inspired with something else, we can always rename it later if we find something better !